benches/bench.rs

```diff
@@ -62,8 +62,16 @@ fn impl_bench_parsing_debug_info<R: Reader>(
         let mut cursor = unit.entries(&abbrevs);
         while let Some((_, entry)) = cursor.next_dfs().expect("Should parse next dfs") {
             let mut attrs = entry.attrs();
-            while let Some(attr) = attrs.next().expect("Should parse entry's attribute") {
-                test::black_box(&attr);
+            loop {
+                match attrs.next() {
```

Comment:
What's the point of going from `while let Some` to `loop` + `match`?

Reply:
It avoids a move of the attribute which the compiler fails to optimise away. But maybe this requires more explanation and deserves its own PR.